### PR TITLE
fix panic

### PR DESCRIPTION
**Pull request by aaronjanse**
When exiting vim!

### CODE REVIEW

File: vterm/ops.go [hunk 1]
```diff
@@ -186,12 +186,12 @@ func (v *VTerm) putChar(ch rune, wide bool) {
...
 	}
 
-	if v.Cursor.Y >= len(v.Screen) {
+	for v.Cursor.Y >= len(v.Screen) {
 		v.Screen = append(v.Screen, []ecma48.StyledChar{})
 	}
 	xDiff := v.Cursor.X - len(v.Screen[v.Cursor.Y]) + 1
 	if xDiff > 0 {
-		v.Screen = append(v.Screen, make([]ecma48.StyledChar, xDiff))
+		v.Screen[v.Cursor.Y] = append(v.Screen[v.Cursor.Y], make([]ecma48.StyledChar, xDiff)...)
```
Comment (PotatoParser) [inline vterm/ops.go:194]: Why do I feel like using append will cause bugs XD

File: vterm/ops.go [hunk 2]
```diff
@@ -186,12 +186,12 @@ func (v *VTerm) putChar(ch rune, wide bool) {
 		Style:  v.Cursor.Style,
 	}
 
-	if v.Cursor.Y >= len(v.Screen) {
+	for v.Cursor.Y >= len(v.Screen) {
```
Comment (PotatoParser) [inline vterm/ops.go:189]: Is this an infinite loop? I don't see an exit condition.

**Review by PotatoParser — CHANGES_REQUESTED** [review]
Looks good, though there are some concerns that I have.

File: vterm/ops.go (hunk 1 above)
Comment (aaronjanse) [reply vterm/ops.go:194]: Yeah, append is scary. But it's supposed to be safe when appending something to itself unless slice stuff is going on

File: vterm/ops.go (hunk 2 above)
Comment (aaronjanse) [reply vterm/ops.go]: Oh yeah, agreed, that's not a good solution. I just pushed something that works similar to xDiff

**Review by PotatoParser — APPROVED** [review]
Looks good!

**Pull request merged**